modules/tpmnBidAdapter.js

```diff
@@ -1,126 +1,222 @@
-    if (bidRequests.length === 0) {
-      return [];
+  onBidWon: function (bid) {
+    if (bid.burl) {
+      utils.triggerPixel(bid.burl);
     }
-    const bids = bidRequests.map(bidToRequest);
-    const bidderApiUrl = URL;
-    const payload = {
-      'bids': [...bids],
-      'site': createSite(bidderRequest.refererInfo)
-    };
-    return [{
-      method: 'POST',
-      url: bidderApiUrl,
-      data: payload
-    }];
+  }
+}
+
+function isBidRequestValid(bid) {
+  return (isValidInventoryId(bid) && (isValidBannerRequest(bid) || isValidVideoRequest(bid)));
+}
+
+function isValidInventoryId(bid) {
+  return 'params' in bid && 'inventoryId' in bid.params && utils.isNumber(bid.params.inventoryId);
+}
+
+function isValidBannerRequest(bid) {
+  const bannerSizes = utils.deepAccess(bid, `mediaTypes.${BANNER}.sizes`);
+  return utils.isArray(bannerSizes) && bannerSizes.length > 0 && bannerSizes.every(size => utils.isNumber(size[0]) && utils.isNumber(size[1]));
+}
+
+function isValidVideoRequest(bid) {
+  const videoSizes = utils.deepAccess(bid, `mediaTypes.${VIDEO}.playerSize`);
+  const videoMimes = utils.deepAccess(bid, `mediaTypes.${VIDEO}.mimes`);
+
+  const isValidVideoSize = utils.isArray(videoSizes) && videoSizes.length > 0 && videoSizes.every(size => utils.isNumber(size[0]) && utils.isNumber(size[1]));
+  const isValidVideoMimes = utils.isArray(videoMimes) && videoMimes.length > 0;
+  return isValidVideoSize && isValidVideoMimes;
+}
+
+function buildRequests(validBidRequests, bidderRequest) {
+  if (validBidRequests.length === 0 || !bidderRequest) return [];
+  let bannerBids = validBidRequests.filter(bid => utils.deepAccess(bid, 'mediaTypes.banner'));
+  let videoBids = validBidRequests.filter(bid => utils.deepAccess(bid, 'mediaTypes.video'));
+  let requests = [];
+
+  bannerBids.forEach(bid => {
+    requests.push(createRequest([bid], bidderRequest, BANNER));
+  });
+
+  videoBids.forEach(bid => {
+    requests.push(createRequest([bid], bidderRequest, VIDEO));
+  });
+
+  return requests;
+}
+
+function createRequest(bidRequests, bidderRequest, mediaType) {
+  const rtbData = CONVERTER.toORTB({ bidRequests, bidderRequest, context: { mediaType } })
+
+  const bid = bidRequests.find((b) => b.params.inventoryId)
+  // console.log('createRequest : ', mediaType, bid);
+
+  if (!rtbData.site) rtbData.site = {}
+  rtbData.site = createSite(bidderRequest.refererInfo)
+
+  if (bidderRequest.gdprConsent) {
+    if (!rtbData.user) rtbData.user = {};
+    if (!rtbData.user.ext) rtbData.user.ext = {};
+    if (!rtbData.regs) rtbData.regs = {};
+    if (!rtbData.regs.ext) rtbData.regs.ext = {};
+    rtbData.user.ext.consent = bidderRequest.gdprConsent.consentString;
+    rtbData.regs.ext.gdpr = bidderRequest.gdprConsent.gdprApplies ? 1 : 0;
+  }
+  if (bid.params.inventoryId) rtbData.ext = {};
+  if (bid.params.inventoryId) rtbData.ext.inventoryId = bid.params.inventoryId
+  if (bid.params.bcat) rtbData.bcat = bid.params.bcat;
```

Comment:
you have accept openrtb specification of bcat as well, you can take your params as an override

Reply:
[da8e928](https://github.com/prebid/Prebid.js/pull/10611/commits/da8e9286419f7b9a254f9bdd0ca79ea8490d5acc)

update openrtb block attributes